idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/purity/PurityPlugin.kt

```diff
@@ -29,12 +33,44 @@ val IdeMetaPlugin.purity: Plugin
         isApplicable = { f: KtNamedFunction ->
           f.nameIdentifier != null && !f.hasModifier(KtTokens.SUSPEND_KEYWORD) &&
             f.resolveToDescriptorIfAny()?.run {
-              !isSuspend && !isSynthesized && !isSuspendLambdaOrLocalFunction() && (ReturnsCheck.ReturnsUnit.check(this) || ExtendedReturnsCheck.ReturnsNothing.check(this)
-                || ExtendedReturnsCheck.ReturnsNullableNothing.check(this))
+              !isSuspend && !isSynthesized && !isSuspendLambdaOrLocalFunction() &&
+                returns(f) { impureTypes }
             } == true
         },
         level = HighlightDisplayLevel.ERROR,
         groupPath = ArrowPath + "PurityPlugin"
+      ),
+      addApplicableInspection(
+        defaultFixText = "SuspendProperty",
+        inspectionHighlightType = { ProblemHighlightType.ERROR },
+        kClass = KtProperty::class.java,
+        highlightingRange = { prop -> prop.nameIdentifierTextRangeInThis() },
+        inspectionText = { prop -> "Property:${prop.name} should be suspended" },
+        applyTo = { prop, project, editor ->
+          prop.initializer?.let { body: KtExpression ->
+            modify(body) { b: KtExpression ->
+              b.unwrapBlockOrParenthesis()
+              // createExpressionByPattern("suspend { $0 }", b)
+            }
+          }
+        },
+        isApplicable = { prop: KtProperty ->
+          prop.resolveToDescriptorIfAny()?.run {
+            !returns(
+              f = { t: KotlinType -> t.constructor },
```

Comment:
This defines Type equality when the TypeConstructor is the same

Reply:
I don't understand this API. Do you mind providing more context on how this works?